library/core/src/primitive_docs.rs

```diff
@@ -731,12 +731,59 @@ mod prim_array {}
 /// assert_eq!(2 * pointer_size, std::mem::size_of::<Box<[u8]>>());
 /// assert_eq!(2 * pointer_size, std::mem::size_of::<Rc<[u8]>>());
 /// ```
+///
+/// ## Trait Implementations
+///
+/// There are several implementations of common traits for slices. Some examples
+/// include:
+///
+/// * [`Clone`]
+/// * [`Eq`], [`Ord`] - for slices whose element type are [`Eq`] or [`Ord`].
+/// * [`Hash`] - for slices whose element type is [`Hash`].
+///
+/// ## Iteration
+///
+/// The slices implement `IntoIterator`. The iterator yields references to the
+/// slice elements.
+///
+/// ```
+/// let numbers = &[0, 1, 2];
+/// for n in numbers {
+///     println!("{} is a number!", n);
+/// }
+/// ```
+///
+/// The mutable slice yields mutable references to the elements:
+///
+/// ```
+/// let mut scores = [7, 8, 9];
+/// for score in &mut scores[..] {
+///     *score += 1;
+/// }
+/// ```
```

Comment:
Both of these examples work with and without `[..]`, but only the `mut` version uses that now. The first one now iterates over a reference to an array, not over the (unsized) slice.

Maybe these should coerce the type explicitly to a slice, and then iterate over that without any `[..]`?

```suggestion
/// ```
/// let numbers: &[i32] = &[0, 1, 2];
/// for n in numbers {
///     println!("{} is a number!", n);
/// }
/// ```
///
/// The mutable slice yields mutable references to the elements:
///
/// ```
/// let scores: &mut [i32] = &mut [7, 8, 9];
/// for score in scores {
///     *score += 1;
/// }
/// ```
```